tendrl/node_agent/discovery/sds/plugins/discover_ceph_sds_details.py

```diff
@@ -25,26 +26,15 @@ def discover_storage_system(self):
             return ret_val
 
         if out:
-            details = json.loads(out)
+            details = out.split()
 
-            ret_val['pkg_version'] = details['version'].split()[2]
-            ret_val['pkg_name'] = details['version'].split()[0]
+            ret_val['pkg_version'] = details[2]
+            ret_val['pkg_name'] = details[0]
 
             # get the cluster_id details
-            cmd = subprocess.Popen(
-                "ceph -s -f json",
-                shell=True,
-                stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE
-            )
-            out, err = cmd.communicate()
-            if err:
-                LOG.error("Error getting cluster details")
-                return ret_val
-            if out:
-                details = json.loads(out)
-                ret_val['detected_cluster_id'] = details['fsid']
-                ret_val['cluster_attrs'] = {'fsid': details['fsid'],
-                                            'name': 'ceph'}
+            raw_data = ini2json.ini_to_dict("/etc/ceph/ceph.conf")
```

Comment:
@nthomas-redhat so while sds detection there is no truth value from where we can get the cluster name. Without knowing the cluster name, how we think of loading the conf file here?

I feel underlying cluster could always be named `ceph` and console can have custom names known to it just for differentiating between multiple ceph clusters. Behind the scene always `fsid` is used for communicating with underlying cluster.

I certainly feel that looking for `/etc/ceph/*.conf` might be problematic at certain stage if ceph decides to have some other files in the same path later.

@Tendrl/tendrl-core comments?